client/network/sync/src/blocks.rs

```diff
@@ -193,6 +193,10 @@ impl<B: BlockT> BlockCollection<B> {
 					let len = (blocks.len() as u32).into();
 					prev = start + len;
 					// Remove all elements from `blocks` and add them to `ready`
+					if let Some(BlockData { block, .. }) = blocks.first() {
+						self.queued_blocks
+							.insert(block.hash, (start, start + (blocks.len() as u32).into()));
```

Comment:
IMHO this should be documented. The "problem" is now that `clear_queued` should be called for every hash of `ready`. We already do this, but someone could forget this in the future.